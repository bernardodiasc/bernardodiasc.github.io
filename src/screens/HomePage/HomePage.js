import React, { Component } from 'react'

import Meta from 'components/Meta'
import Layout from 'displays/Layout'
import PostsListing from 'displays/PostsListing'

class HomePage extends Component {
  static defaultProps = {
    posts: [],
  }

  render() {
    return (
      <Layout>
        <Meta />
        <PostsListing items={this.props.posts} />
      </Layout>
    )
  }
}

export default HomePage
