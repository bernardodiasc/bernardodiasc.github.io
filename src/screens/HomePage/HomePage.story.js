import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import HomePage from './HomePage'
import HomePageReadme from './README.md'
storiesOf('Screens/HomePage', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{HomePageReadme}</Docs>)
  .add('Example', () => <HomePage />)
  .add('Minimal render', () => <HomePage />)
