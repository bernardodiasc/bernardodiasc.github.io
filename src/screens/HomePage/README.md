# HomePage

## How to use

```
import HomePage from 'screens/HomePage'
```

```
<HomePage />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
