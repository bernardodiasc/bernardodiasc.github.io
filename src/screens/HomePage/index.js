import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { getAllPostsForListing } from 'selectors/data'

import HomePage from './HomePage'

const mapStateToProps = store => ({
  posts: getAllPostsForListing(store),
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({
  }, dispatch)
)

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(HomePage)
