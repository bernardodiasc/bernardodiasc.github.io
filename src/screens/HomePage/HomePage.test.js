import React from 'react'
import { shallow } from 'enzyme'

import HomePage from './HomePage.js'

it('renders without crashing', () => {
  const component = shallow(<HomePage />)
  expect(component).toMatchSnapshot()
})
