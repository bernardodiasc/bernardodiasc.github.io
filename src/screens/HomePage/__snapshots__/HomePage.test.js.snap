// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Layout
  breadcrumbs={Array []}
>
  <Meta />
  <PostsListing
    items={Array []}
  />
</Layout>
`;
